extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureStorageDruidModule.java

```diff
@@ -137,9 +135,10 @@ public AzureClientFactory getAzureClientFactory(final AzureAccountConfig config)
   @Provides
   @LazySingleton
```

Comment:
This seems to be providing the storage used for the segments in particular, is that right? Given this should be maybe provide this with an annotation, so that other existing / future consumers of the AzureStorage that have the AzureStorage instance injected are clear in their intention to use the segment storage location rather than ingest location?